viadot/sources/sap_rfc.py

```diff
@@ -84,6 +85,103 @@ def trim_where(where: str) -> Tuple[str, OrderedDictType[str, str]]:
     return where_trimmed, wheres_to_add
 
 
+def detect_extra_rows(
+    row_index: int, data_raw: np.array, chunk: int, fields: List[str]
+) -> Union[int, np.array, bool]:
+    """check if, in between calls to the SAP table, the number of rows have increased.
+        If so, remove the last rows added, to fit the size of the previous columns.
+
+    Args:
+        row_index (int): Number of rows set it down in he first iteration with the SAP table.
+        data_raw (np.array): Array with the data retrieve from SAP table.
+        chunk (int): The part number in which a number of SAP table columns have been split.
+        fields (List[str]): A list with the names of the columns in a chunk.
+
+    Returns:
+        Union[int, np.array, bool]: A tuple with the parameters "row_index", "data_raw", and a new
+            boolean variable "cont" to indicate when the for loop has to be restarted.
+    """
+    cont = False
+    if row_index == 0:
+        row_index = data_raw.shape[0]
+        if row_index == 0:
+            logger.warning(
+                f"Empty output was generated for chunk {chunk} in columns {fields}."
+            )
+            chunk += 1
+            cont = True
+    elif data_raw.shape[0] != row_index:
+        data_raw = data_raw[:row_index]
+        logger.warning(
+            f"New rows were generated during the execution of the script. The table is truncated to the number of rows for the first chunk"
+        )
+
+    return row_index, data_raw, cont
+
+
+def catch_extra_separators(
+    data_raw: np.array, record_key: str, sep: str, fields: List[str], replacement: str
+) -> np.array:
+    """Function to replace extra separators in every row of the
+
+    Args:
+        data_raw (np.array): Array with the data retrieve from SAP table.
+        record_key (str): Key word to extract the data from the numpy array "data_raw".
+        sep (str): Which separator to use when querying SAP.
+        fields (List[str]): A list with the names of the columns in a chunk.
+        replacement (str): In case of sep is on a columns, set up a new character to replace
+            inside the string to avoid flow breakdowns.
+
+    Returns:
+        np.array: The argument "data_raw" with no extra delimiters.
+    """
+
+    # remove scape characters from data_raw ("\t")
+    for n, r in enumerate(data_raw):
+        if "\t" in r[record_key]:
+            data_raw[n][record_key] = r[record_key].replace("\t", " ")
+
+    # first we identify where the data has an extra separator in text columns.
+    sep_counts = np.array([], dtype=int)
+    for row in data_raw:
+        sep_counts = np.append(sep_counts, row[record_key].count(f"{sep}"))
```

Comment:
In different files we mostly use pandas,would you be able to replace numpy with pandas?

Reply:
Not all the functionalities implemented with `numpy` can be done with `pandas`. For now, I don't think that is necessary, furthermore, it is faster.